### PR TITLE
Add case-insensitive search to clp-s source code and the package

### PR DESCRIPTION
**Pull request by wraymo**
# References

# Description

This PR adds case-insensitive search to clp-s souce code and the package

# Validation performed

+ Compiled clp-s and compressed a portion of [mongodb](https://zenodo.org/records/10516285) dataset.
+ Executed two queries, `msg: "Initialized Wire Specification"` (for logtype dict test) and `c: repl` (for var dict test). These queries yielded no results without the `--ignore-case` flag but returned results when the flag was applied.
+ Built the package and compressed [hive-24hrs](https://zenodo.org/records/7094921#.Y5JbH33MKHs) dataset using the package
+ Running `sbin/search.sh "org.apache.hadoop.mapred.Task: using resourcecalculatorprocesstree : [ ]"` resulted in no results, while `sbin/search.sh --ignore-case "org.apache.hadoop.mapred.Task: using resourcecalculatorprocesstree : [ ]"` returned relevant results.

### CODE REVIEW

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]
Can we also add an example to the docs?

**Comment by wraymo** [comment]
> Can we also add an example to the docs?

Do I need to add examples for both clp and clp-s? Add this option to an existing example, or create a new one?

**Comment by kirkrodrigues** [comment]
> Do I need to add examples for both clp and clp-s? Add this option to an existing example, or create a new one?

Good point. Yeah, we should probably add an example for both clp and clp-s.

**Comment by wraymo** [comment]
> > Do I need to add examples for both clp and clp-s? Add this option to an existing example, or create a new one?
> 
> Good point. Yeah, we should probably add an example for both clp and clp-s.

But for clp, a lot of options are not shown with an example in the doc. If we want to add this one, what about `compression-level`, `target-encoded-file-size`, `remove-path-prefix`, etc?

**Comment by kirkrodrigues** [comment]
> But for clp, a lot of options are not shown with an example in the doc. If we want to add this one, what about `compression-level`, `target-encoded-file-size`, `remove-path-prefix`, etc?

I think `--ignore-case` would probably be the most used option out of all the ones you mentioned since folks would forget to use capitals when entering a query. It's also a hint to users that by default the query is case-sensitive.

**Comment by kirkrodrigues** [comment]
Fwiw, we should indeed document all the other parameters, but that will be in another reference doc once we have the rest of CLP's docs set up online.

File: components/core/src/clp_s/clp-s.cpp [hunk 1]
```diff
@@ -329,6 +330,4 @@ int main(int argc, char const* argv[]) {
             }
         }
     }
-
-    return 0;
```
Comment (kirkrodrigues) [inline components/core/src/clp_s/clp-s.cpp:333]: Why delete this?

File: docs/core/clp-structured.md [hunk 2]
```diff
@@ -113,10 +113,11 @@ or
 ./clp-s s /mnt/data/archives1 'level: ERROR AND message: "job*"'
 ```
 
-**Find both FATAL and ERROR log events:**
+**Find both FATAL and ERROR log events and ignore case distinctions between values in the query and
+the compressed data:**
```
Comment (kirkrodrigues) [inline docs/core/clp-structured.md:117]:
```suggestion
**Find FATAL or ERROR log events and ignore case distinctions between values in the query and the
compressed data:**
```

File: docs/core/clp-unstructured.md [hunk 3]
```diff
@@ -91,10 +91,11 @@ Usage:
 
 ### Examples
 
-**Search `/mnt/data/archives1` for specific ERROR logs:**
+**Search `/mnt/data/archives1` for specific ERROR logs and ignore case distinctions between values
+in the query and the compressed data:**
```
Comment (kirkrodrigues) [inline docs/core/clp-unstructured.md:95]:
```suggestion
**Search `/mnt/data/archives1` for specific ERROR logs and ignore case distinctions:**
```

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: components/core/src/clp_s/clp-s.cpp (hunk 1 above)
Comment (wraymo) [reply components/core/src/clp_s/clp-s.cpp:333]: It's by mistake

**Review by kirkrodrigues — APPROVED** [review]
Commit msg: `Add support for case-insensitive searches in clp-s and the package. (#272)`?

**Pull request merged**